modules/core/src/smithy4s/Validator.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+sealed trait Validator[A, B] {
+  def validate(value: A): Either[String, B]
+}
+
+object Validator {
+
+  def of[A, B]: PartialValidatorBuilder[A, B] =
+    new PartialValidatorBuilder[A, B]
+
+  private[smithy4s] class PartialValidatorBuilder[A, B] {
```

Comment:
this always throws me off, because I'm not quite sure it's possible to call its methods from outside the smithy4s package. 

I'd rather make the constructor private : `class PartialValidatorBuilder[A, B] private ()`

Reply:
Done.